### DISCUSSION TITLE
I keep getting these errors with _app.js

### DISCUSSION THREAD

**Discussion by taylor-lindores-reeves** [vercel/next.js#13664] [Help]
<img width="1052" alt="Screenshot 2020-06-02 13 46 42" src="https://user-images.githubusercontent.com/27443324/83521886-c64bc600-a4d7-11ea-8b42-708cb8162336.png">

The code I am running in the getInitialProps of _app.js is sending a request to the server and returning the user. In Sentry, the error is coming from /api/user request.

_Here is the getInitialProps in _app.js_
```
static async getInitialProps({ Component, router, ctx }) {
    const res = await fetch(`${environments.PUBLIC_URL}/api/user`, {
      method: "POST",
      headers: {
        Accept: "application/json, text/plain, */*",
        "Content-Type": "application/json",
      },
    }).then((res) => res.json());

    console.log(res);

    let pageProps = {
      user: res,
    };

    if (Component.getInitialProps) {
      pageProps = await Component.getInitialProps(ctx);
    }

    return { pageProps };
  }

```

And here is the api code in authController.js_

```
exports.user = async (req, res, next) => {
  if (req.cookies.token) {
    const decoded = jwt.verify(req.cookies.token, process.env.JWT_SECRET);
    if (decoded.userId) {
      const user = await User.findById(decoded.userId);
      return res.json({
        id: decoded.userId,
        name: user.first_name,
        last_name: user.last_name,
        email: user.email,
      });
    }
  }
  return res.status("200").json(null);
};
```

Now if anyone can help me figure out how to do this more elegantly and error free, it would REALLY help because people are actually paying for a service on my website which is now giving them 404 errors when the response is...

<img width="1052" alt="Screenshot 2020-06-02 13 46 42" src="https://user-images.githubusercontent.com/27443324/83522154-2cd0e400-a4d8-11ea-9a4a-5cf677b50904.png">

If you need any more code or anything, please let me know.

**Comment by jamesmosier** [comment]
It may help if you add a `catch` to your `fetch` request:

```js
try {
  const response = await fetch(`${environments.PUBLIC_URL}/api/user`, {
    method: "POST",
    headers: {
      Accept: "application/json, text/plain, */*",
      "Content-Type": "application/json",
    },
  })
  const data = await response.json();
  return data;
} catch (e) {
  console.log(e);
  // log to sentry
}
```

I have seen that error most often when an API is returning an error.
